src/http/status.cr

```diff
@@ -0,0 +1,163 @@
+  def informational?
+    100 <= value <= 199
+  end
+
+  # Returns `true` if the response status code is between 200 and 299.
+  def success?
+    200 <= value <= 299
+  end
+
+  # Returns `true` if the response status code is between 300 and 399.
+  def redirection?
+    300 <= value <= 399
+  end
+
+  # Returns `true` if the response status code is between 400 and 499.
+  def client_error?
+    400 <= value <= 499
+  end
+
+  # Returns `true` if the response status code is between 500 and 599.
+  def server_error?
+    500 <= value <= 599
+  end
+
+  # Returns the default status description of the given HTTP status code.
+  def description : String
+    case value
+    when 100 then "Continue"
+    when 101 then "Switching Protocols"
+    when 102 then "Processing"
+    when 200 then "OK"
+    when 201 then "Created"
+    when 202 then "Accepted"
+    when 203 then "Non-Authoritative Information"
+    when 204 then "No Content"
+    when 205 then "Reset Content"
+    when 206 then "Partial Content"
+    when 207 then "Multi-Status"
+    when 208 then "Already Reported"
+    when 226 then "IM Used"
+    when 300 then "Multiple Choices"
+    when 301 then "Moved Permanently"
+    when 302 then "Found"
+    when 303 then "See Other"
+    when 304 then "Not Modified"
+    when 305 then "Use Proxy"
+    when 306 then "Switch Proxy"
+    when 307 then "Temporary Redirect"
+    when 308 then "Permanent Redirect"
+    when 400 then "Bad Request"
+    when 401 then "Unauthorized"
+    when 402 then "Payment Required"
+    when 403 then "Forbidden"
+    when 404 then "Not Found"
+    when 405 then "Method Not Allowed"
+    when 406 then "Not Acceptable"
+    when 407 then "Proxy Authentication Required"
+    when 408 then "Request Timeout"
+    when 409 then "Conflict"
+    when 410 then "Gone"
+    when 411 then "Length Required"
+    when 412 then "Precondition Failed"
+    when 413 then "Payload Too Large"
+    when 414 then "URI Too Long"
+    when 415 then "Unsupported Media Type"
+    when 416 then "Range Not Satisfiable"
+    when 417 then "Expectation Failed"
+    when 418 then "I'm a teapot"
+    when 421 then "Misdirected Request"
+    when 422 then "Unprocessable Entity"
+    when 423 then "Locked"
+    when 424 then "Failed Dependency"
+    when 426 then "Upgrade Required"
+    when 428 then "Precondition Required"
+    when 429 then "Too Many Requests"
+    when 431 then "Request Header Fields Too Large"
+    when 451 then "Unavailable For Legal Reasons"
+    when 500 then "Internal Server Error"
+    when 501 then "Not Implemented"
+    when 502 then "Bad Gateway"
+    when 503 then "Service Unavailable"
+    when 504 then "Gateway Timeout"
+    when 505 then "HTTP Version Not Supported"
+    when 506 then "Variant Also Negotiates"
+    when 507 then "Insufficient Storage"
+    when 508 then "Loop Detected"
+    when 510 then "Not Extended"
+    when 511 then "Network Authentication Required"
```

Comment:
You can use this macro:
```cr
enum HTTP::Status
  Continue           = 100
  SwitchingProtocols = 101
  Processing         = 102
  EarlyHints         = 103

  def description
    {% begin %}
     case value
     {% for member in @type.constants %}
     when {{member.id}}.value then "{{member.gsub(/([A-Z])/, " \\1")[1..-1]}}"
     {% end %}
     else ""
     end
    {% debug %}
    {% end %}
  end
end
puts HTTP::Status::EarlyHints.description #=> Early Hints
```
If others think that the dash matter, a gsub could be added to replace `_` to `-`, same for `Im ` to `I'm `,

Reply:
This is really not a good use case for macros. There is no effective benefit. It unnecessarily makes the code more complex and increases compiler work.
Also it's nice when you can grep for the message string.

Reply:
macro aren't designed to reduce boilerplate?

Reply:
This also prevent https://github.com/crystal-lang/crystal/pull/7231#discussion_r244495514 . I don't see a use case for grep, you can search in the enum values directly.

Reply:
You shouldn't try to rewrite something into a macro, when plain code is perfectly fine.